src/ErrorProne.NET.CoreAnalyzers/AsyncAnalyzers/ConfigureAwaitRequiredAnalyzer.cs

```diff
@@ -27,38 +26,56 @@ public ConfigureAwaitRequiredAnalyzer()
         /// <inheritdoc />
         protected override void InitializeCore(AnalysisContext context)
         {
-            context.RegisterSyntaxNodeAction(AnalyzeAwaitExpression, SyntaxKind.AwaitExpression);
+            context.RegisterCompilationStartAction(context =>
+            {
+                var compilation = context.Compilation;
+
+                var configureAwaitConfig = ConfigureAwaitConfiguration.TryGetConfigureAwait(context.Compilation);
+                if (configureAwaitConfig != ConfigureAwait.UseConfigureAwaitFalse)
```

Comment:
All callbacks are checking for this. So, we can actually check it once in compilation start, and not register any action at all when not needed.